File: src/test/java/com/fluxchess/pulse/ColorTest.java
/*
 * Copyright (C) 2013-2019 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

import org.junit.Test;

import static com.fluxchess.pulse.Color.*;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class ColorTest {

	@Test
	public void testValues() {
		for (int color : Color.values) {
			assertThat(Color.values[color], is(color));
		}
	}

	@Test
	public void testOpposite() {
		assertThat(opposite(BLACK), is(WHITE));
		assertThat(opposite(WHITE), is(BLACK));
	}
}

File: src/test/cpp/colortest.cpp
/*
 * Copyright (C) 2013-2016 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include "color.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(colortest, testValues) {
	for (auto color : Color::values) {
		EXPECT_EQ(color, Color::values[color]);
	}
}

TEST(colortest, testOpposite) {
	EXPECT_EQ(+Color::WHITE, Color::opposite(Color::BLACK));
	EXPECT_EQ(+Color::BLACK, Color::opposite(Color::WHITE));
}
